reviewing/plugin-commands-listing/test/index.ts

```diff
@@ -197,3 +197,118 @@ pkg@1.0.0 ${pkgDir}
 dependencies:
 dep file:../dep`)
 })
+
+test('listing packages with --lockfile-only (without node_modules)', async () => {
+  prepare({
+    dependencies: {
+      'is-positive': '1.0.0',
+    },
+    devDependencies: {
+      'is-negative': '1.0.0',
+    },
+  })
+
+  await execa('node', [pnpmBin, 'install'])
+  await execa('rm', ['-rf', 'node_modules'])
+
+  {
+    const output = await list.handler({
+      dev: false,
+      dir: process.cwd(),
+      lockfileOnly: true,
+      optional: false,
+      virtualStoreDirMaxLength: process.platform === 'win32' ? 60 : 120,
+    }, [])
+
+    expect(stripAnsi(output)).toBe(`Legend: production dependency, optional only, dev only
+
+project@0.0.0 ${process.cwd()}
+
+dependencies:
+is-positive 1.0.0`)
+  }
+
+  {
+    const output = await list.handler({
+      dir: process.cwd(),
+      lockfileOnly: true,
+      optional: false,
+      production: false,
+      virtualStoreDirMaxLength: process.platform === 'win32' ? 60 : 120,
+    }, [])
+
+    expect(stripAnsi(output)).toBe(`Legend: production dependency, optional only, dev only
+
+project@0.0.0 ${process.cwd()}
+
+devDependencies:
+is-negative 1.0.0`)
+  }
+
+  {
+    const output = await list.handler({
+      dir: process.cwd(),
+      lockfileOnly: true,
+      virtualStoreDirMaxLength: process.platform === 'win32' ? 60 : 120,
+    }, [])
+
+    expect(stripAnsi(output)).toBe(`Legend: production dependency, optional only, dev only
+
+project@0.0.0 ${process.cwd()}
+
+dependencies:
+is-positive 1.0.0
+
+devDependencies:
+is-negative 1.0.0`)
+  }
+})
+
+test('listing packages with --lockfile-only in JSON format', async () => {
+  prepare({
+    dependencies: {
+      'is-positive': '1.0.0',
+    },
+  })
+
+  await execa('node', [pnpmBin, 'install'])
+  await execa('rm', ['-rf', 'node_modules'])
```

Comment:
use the `@zkochan/rimraf` library to remove node_modules. Use the sync function from it.